### ISSUE TITLE
d3-interpolate depends on vulnerable d3-color

### ISSUE THREAD

**Issue by alexvanacker** [plotly/plotly.js#6333]
Following the vulnerability revealed in https://security.snyk.io/vuln/SNYK-JS-D3COLOR-1076592

plotly.js uses d3-interpolate 1.4.1 which still depends on vulnerable d3-color 1.4.1

Given plotly currenlty depends on d3-color 3.8 (see https://github.com/plotly/d3/blob/master/package.js) which is not affected, could we update this dependency to 3.0.1?

**Comment by whittet** [comment]
Impacts angular 14

**Comment by krisstern** [comment]
The vulnerability in `d3-color` was patched for version  >=3.1.0.

**Comment by peratik** [comment]
![image](https://user-images.githubusercontent.com/6568353/195119439-17e6193e-95c4-4030-a7aa-2534dbb6afe0.png)

`"plotly.js": "^2.15.1",`

Hi, Same here, Please advice.

**Comment by archmoj** [comment]
The `d3-interpolate` module is an ES module.
To handle that we need to change our build processes quite a bit. 
This may include migrating from `browserify` to `webpack`.
.AND. we are currently investigating that.

**Comment by raulmarindev** [comment]
They started with the migration to `webpack`. https://github.com/plotly/plotly.js/pull/6355

**Comment by whittet** [comment]
Looks like now that webpack is merged to master the d3-interpolate package version can be updated to 3.0.1. I asked for same in a comment on this PR: https://github.com/plotly/plotly.js/pull/6344

**Comment by krisstern** [comment]
I checked that for a fix we will need `d3-interpolate` version of at least `3.1.0`, not `3.0.1`.

**Comment by flying-sheep** [comment]
I did that in #6427, but I don’t know how to fix the build error.

If you want this fixed, feel free to help fix that PR.

**Comment by krisstern** [comment]
I found out if I use `yarn` I can add the following to `package.json` and then run `yarn install` (may need to delete the entire `node_modules` beforehand) to get rid of the vulnerability after upgrading `plotly.js` to version 2.17.1:
```JavaScript
"resolutions": {
    "plotly.js/d3-interpolate/d3-color": "3.1.0"
}
```

**Comment by whittet** [comment]
@krisstern Thanks! I don't have yarn, so my alternative is to upgrade npm to ^8.3.0 and node to ^16.13.0. Add following to package.json:

  "overrides": {
    "d3-color": "3.1.0",
  }

Although, this completely rewrote the angular site's package-lock.json.

**Comment by archmoj** [comment]
Addressed in #6463.
